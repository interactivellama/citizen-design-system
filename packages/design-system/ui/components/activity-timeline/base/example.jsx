// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from 'ui/components/button-icons/base/example';
import { ActivityTimelineItem as Event } from 'ui/components/activity-timeline/event/example';
import { ActivityTimelineItem as Call } from 'ui/components/activity-timeline/call/example';
import { ActivityTimelineItem as Email } from 'ui/components/activity-timeline/email/example';
import { ActivityTimelineItem as Task } from 'ui/components/activity-timeline/task/example';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

export let TimelineIcon = props =>
  <div
    className={classNames(
      'slds-icon_container',
      props.symbol == 'log_a_call' ?
        'slds-icon-standard-log-a-call'
      : 'slds-icon-standard-' + props.symbol
    )}
    title={props.symbol == 'log_a_call' ? 'call' : props.symbol }
  >
    <SvgIcon
      className="slds-icon slds-icon--small"
      sprite="standard"
      symbol={ props.symbol }
    />
  </div>;

export let TimelineActions = props =>
  <div className="slds-timeline__actions">
    <p className="slds-timeline__date">{ props.date || 'Feb 24' }</p>
    <ButtonIcon
      className="slds-button--icon-border-filled slds-button--icon-x-small"
      symbol="down"
      assistiveText={ props.assistiveText || 'More Options' }
      aria-haspopup="true"
      title={ props.title || 'More Options' }
    />
  </div>;

export default (
  <ul className="slds-timeline">
    <li>
      <Task />
    </li>
    <li>
      <Event />
    </li>
    <li>
      <Call />
    </li>
    <li>
      <Email />
    </li>
  </ul>
);
