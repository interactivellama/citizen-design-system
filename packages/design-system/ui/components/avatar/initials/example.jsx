// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default (
  <Avatar>
    <AvatarInitials
      iconClassName="slds-icon-standard-account"
      title="company name"
      initials="Ac"
    />
  </Avatar>
);

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="javascript:void(0);">
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
