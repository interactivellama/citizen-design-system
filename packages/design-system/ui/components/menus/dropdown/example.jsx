// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from '../../button-icons/base/example';
import SvgIcon from '../../../shared/svg-icon';
import classNames from 'classnames';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

let Demo = props =>
  <div className="demo-only" {...props}>
    {props.children}
  </div>;

const TableIcon = (
  <SvgIcon className="slds-icon slds-icon--x-small slds-icon-text-default slds-m-left--small slds-shrink-none" sprite="utility" symbol="table" />
);

const KanbanIcon = (
  <SvgIcon className="slds-icon slds-icon--x-small slds-icon-text-default slds-m-left--small slds-shrink-none" sprite="utility" symbol="kanban" />
);

const ListIcon = (
  <SvgIcon className="slds-icon slds-icon--x-small slds-icon-text-default slds-m-left--small slds-shrink-none" sprite="utility" symbol="side_list" />
);

export let Trigger = props =>
  <div className={classNames('slds-dropdown-trigger slds-dropdown-trigger--click', props.className)}>
    { props.triggerIcon ? props.triggerIcon : <ButtonIcon className={classNames(props.inverse ? 'slds-button--icon-border-inverse' : 'slds-button--icon-border-filled')} symbol="down" assistiveText="Show More" title="Show More" aria-haspopup="true" /> }
    {props.children}
  </div>;

export let Menu = props =>
  <div className={classNames('slds-dropdown', props.className)}>
    {props.children}
  </div>;

export let MenuList = props =>
  <ul className={classNames('slds-dropdown__list', props.className)} role="menu">
    {props.children}
  </ul>;

export let MenuItem = props => {
  const {
    className,
    isSelectable,
    isSelected,
    children,
    iconRight,
    tabIndex,
    ...rest
  } = props;
  let ariaChecked;

  if (isSelectable) {
    ariaChecked = isSelected ? 'true' : 'false';
  }

  return (
    <li {...rest} className={classNames('slds-dropdown__item', className)} role="presentation">
      <a
        href="javascript:void(0);"
        role={isSelectable ? 'menuitemcheckbox' : 'menuitem'}
        aria-checked={ariaChecked}
        tabIndex={tabIndex || '-1'}>
        <span className="slds-truncate" title={children}>
          { isSelectable ? <SvgIcon
            className="slds-icon slds-icon--selected slds-icon--x-small slds-icon-text-default slds-m-right--x-small"
            sprite="utility"
            symbol="check" /> : null }
          { children }
        </span>
        { iconRight || null }
      </a>
    </li>
  );
};

/// ///////////////////////////////////////////
// State Constructor(s)
/// ///////////////////////////////////////////

let Default = props =>
  <Demo style={{height: '220px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left">
        <MenuList>
          <MenuItem tabIndex="0">Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
          <MenuItem>Menu Item Three</MenuItem>
          <li className="slds-has-divider--top-space" role="separator" />
          <MenuItem>Menu Item Four</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let Small = props =>
  <Demo style={{height: '220px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--small">
        <MenuList>
          <MenuItem tabIndex="0">Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
          <MenuItem>Menu Item Three</MenuItem>
          <li className="slds-has-divider--top-space" role="separator" />
          <MenuItem>Menu Item Four</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let Medium = props =>
  <Demo style={{height: '220px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--medium">
        <MenuList>
          <MenuItem tabIndex="0">Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
          <MenuItem>Menu Item Three</MenuItem>
          <li className="slds-has-divider--top-space" role="separator" />
          <MenuItem>Menu Item Four</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let Large = props =>
  <Demo style={{height: '220px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--large">
        <MenuList>
          <MenuItem tabIndex="0">Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
          <MenuItem>Menu Item Three</MenuItem>
          <li className="slds-has-divider--top-space" role="separator" />
          <MenuItem>Menu Item Four</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let SubHeader = props =>
  <Demo style={{height: '260px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--small">
        <MenuList>
          <li className="slds-dropdown__header slds-truncate" title="Menu Sub Heading" role="separator">
            <span className="slds-text-title--caps">Menu Sub Heading</span>
          </li>
          <MenuItem tabIndex="0">Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
          <li className="slds-dropdown__header slds-truncate" title="Menu Sub Heading" role="separator">
            <span className="slds-text-title--caps">Menu Sub Heading</span>
          </li>
          <MenuItem>Menu Item One</MenuItem>
          <MenuItem>Menu Item Two</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let IconLeft = props =>
  <Demo style={{height: '170px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--small">
        <MenuList>
          <MenuItem className="slds-is-selected" isSelected="true" isSelectable tabIndex="0">Menu Item One</MenuItem>
          <MenuItem isSelectable>Menu Item Two</MenuItem>
          <MenuItem isSelectable>Menu Item Three</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let DoubleIcon = props =>
  <Demo style={{height: '170px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--small">
        <MenuList>
          <MenuItem className="slds-is-selected" isSelected="true" isSelectable iconRight={TableIcon} tabIndex="0">Table View</MenuItem>
          <MenuItem isSelectable iconRight={KanbanIcon}>Kanban Board</MenuItem>
          <MenuItem isSelectable iconRight={ListIcon}>List View</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

let IconRight = props =>
  <Demo style={{height: '170px'}}>
    <Trigger className="slds-is-open">
      <Menu className="slds-dropdown--left slds-dropdown--small">
        <MenuList>
          <MenuItem iconRight={TableIcon} tabIndex="0">Table View</MenuItem>
          <MenuItem iconRight={KanbanIcon}>Kanban Board</MenuItem>
          <MenuItem iconRight={ListIcon}>List View</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </Demo>;

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export const Context = props =>
  <div style={{height: '260px'}}>
    {props.children}
  </div>;

export default (
  <Trigger className="slds-is-open">
    <Menu className="slds-dropdown--left">
      <MenuList>
        <MenuItem tabIndex="0">Menu Item One</MenuItem>
        <MenuItem>Menu Item Two</MenuItem>
        <MenuItem>Menu Item Three</MenuItem>
        <li className="slds-has-divider--top-space" role="separator" />
        <MenuItem>Menu Item Four</MenuItem>
      </MenuList>
    </Menu>
  </Trigger>
);

export let examples = [
  {
    id: 'dropdown-menu-header',
    label: 'Sub Heading',
    element: <SubHeader />
  },
  {
    id: 'dropdown-menu-icon-left',
    label: 'Single Icon - Left',
    element: <IconLeft />
  },
  {
    id: 'dropdown-menu-icon-double',
    label: 'Double Icon - Left/Right',
    element: <DoubleIcon />
  },
  {
    id: 'dropdown-menu-icon-right',
    label: 'Single Icon - Right',
    element: <IconRight />
  },
  {
    id: 'action-overflow',
    label: 'Overflow menu with actions',
    element:
      <Trigger
        className="slds-is-open"
        triggerIcon={
          <ButtonIcon
            className="slds-button--icon-border-filled slds-button--icon-x-small"
            sprite="utility"
            symbol="down"
            assistiveText="Show More"
            aria-haspopup="true"
            title="Show More"
          />
        }
      >
        <Menu className="slds-dropdown--left slds-dropdown--actions">
          <MenuList>
            <MenuItem tabIndex="0">Action One</MenuItem>
            <MenuItem>Action Two</MenuItem>
            <MenuItem>Action Three</MenuItem>
          </MenuList>
        </Menu>
      </Trigger>
  }
];
