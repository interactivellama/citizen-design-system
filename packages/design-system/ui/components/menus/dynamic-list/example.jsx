// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Listbox, ListboxList, ListboxItem, PickList } from '../../picklist/base/example';
import { Popover } from '../../popovers/base/example';
import { FormElement } from '../../form-element/base/example';
import SvgIcon from '../../../shared/svg-icon';
import _ from 'lodash';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

let Footer = props =>
  <ul>
    <li>
      <button className="slds-button slds-lookup__item-action slds-lookup__item-action--label">
        <SvgIcon className="slds-button__icon slds-button__icon--left" sprite="utility" symbol="add" />
        Favorite this page
      </button>
    </li>
    <li>
      <button className="slds-button slds-lookup__item-action slds-lookup__item-action--label">
        <SvgIcon className="slds-button__icon slds-button__icon--left" sprite="utility" symbol="edit" />
        Edit Favorites
      </button>
    </li>
  </ul>;

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export let states = [
  {
    id: '0-items',
    label: '0 items',
    element:
      <Popover
        className="slds-nubbin--top-right"
        bodyClassName="slds-p-around--medium"
        footerClassName="slds-p-horizontal--none"
        footer={<Footer />}
      >
        <h3 className="slds-text-title--caps slds-m-bottom--x-small" role="presentation">My Favorites</h3>
        <p>You can favorite any page!</p>
      </Popover>
  },
  {
    id: '1-item',
    label: '1 item',
    element:
      <Popover
        className="slds-nubbin--top-right"
        bodyClassName="slds-p-vertical--xx-small slds-p-horizontal--none"
        footerClassName="slds-p-horizontal--none"
        footer={<Footer />}
      >
        <Listbox className="slds-dropdown--length-10">
          <ListboxList role="group" aria-label="My Favorites">
            <ListboxItem role="presentation" headerText="My Favorites" />
            <ListboxItem tabIndex="0">
              <SvgIcon className="slds-icon slds-icon-standard-account slds-icon--small slds-media__figure" sprite="standard" symbol="account" />
              <span className="slds-media__body">
                <span className="slds-lookup__result-text">Salesforce.com, Inc.</span>
                <span className="slds-lookup__result-meta slds-text-body--small">Account &bull; San Francisco</span>
              </span>
            </ListboxItem>
          </ListboxList>
        </Listbox>
      </Popover>
  },
  {
    id: 'less-than-10-items',
    label: '< 10 items',
    element:
      <Popover
        className="slds-nubbin--top-right"
        bodyClassName="slds-p-vertical--xx-small slds-p-horizontal--none"
        footerClassName="slds-p-horizontal--none"
        footer={<Footer />}
      >
        <Listbox className="slds-dropdown--length-10">
          <ListboxList role="group" aria-label="My Favorites">
            <ListboxItem role="presentation" headerText="My Favorites" />
            { _.times(7, i =>
              <ListboxItem tabIndex={(i === 0) ? '0' : null} key={`listbox-${i}`}>
                <SvgIcon className="slds-icon slds-icon-standard-account slds-icon--small slds-media__figure" sprite="standard" symbol="account" />
                <span className="slds-media__body">
                  <span className="slds-lookup__result-text">Salesforce.com, Inc.</span>
                  <span className="slds-lookup__result-meta slds-text-body--small">Account &bull; San Francisco</span>
                </span>
              </ListboxItem>
            )}
          </ListboxList>
        </Listbox>
      </Popover>
  },
  {
    id: '10-plus-items',
    label: '> 10 items',
    element:
      <Popover
        className="slds-nubbin--top-right"
        bodyClassName="slds-p-vertical--xx-small slds-p-horizontal--none"
        footerClassName="slds-p-horizontal--none"
        footer={<Footer />}
      >
        <PickList classNames="slds-size--1-of-1" isOpen>
          <FormElement
            className="slds-p-around--small"
            label="Categories"
            hideLabel
            inputId="text-input-01"
            inputIcon="left"
          >
            <SvgIcon className="slds-input__icon" sprite="utility" symbol="search" />
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="option-list-01"
              autoComplete="off"
              className="slds-input slds-lookup__search-input"
              id="text-input-01"
              placeholder="Quick find&hellip;"
              role="textbox"
              type="text"
            />
          </FormElement>
          <Listbox className="slds-dropdown--length-10" id="option-list-01">
            <ListboxList role="group" aria-label="My Favorites">
              <ListboxItem role="presentation" headerText="My Favorites" />
              { _.times(11, i =>
                <ListboxItem key={`listbox-${i}`}>
                  <SvgIcon className="slds-icon slds-icon-standard-account slds-icon--small slds-media__figure" sprite="standard" symbol="account" />
                  <span className="slds-media__body">
                    <span className="slds-lookup__result-text">Salesforce.com, Inc.</span>
                    <span className="slds-lookup__result-meta slds-text-body--small">Account &bull; San Francisco</span>
                  </span>
                </ListboxItem>
              )}
            </ListboxList>
          </Listbox>
        </PickList>
      </Popover>
  }
];
