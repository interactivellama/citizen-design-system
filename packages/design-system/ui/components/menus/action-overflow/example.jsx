// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from '../../button-icons/base/example';
import { Trigger } from '../dropdown/example';
import { Menu } from '../dropdown/example';
import { MenuList } from '../dropdown/example';
import { MenuItem } from '../dropdown/example';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

const moreIcon = (
  <ButtonIcon
    className="slds-button_icon-border-filled slds-button_icon-x-small"
    sprite="utility"
    symbol="down"
    assistiveText="Show More"
    aria-haspopup="true"
    title="Show More"
  />
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default (
  <div className="demo-only" style={{ height: '165px' }}>
    <Trigger className="slds-is-open" triggerIcon={moreIcon}>
      <Menu className="slds-dropdown_left slds-dropdown_actions">
        <MenuList>
          <MenuItem tabIndex="0">Action One</MenuItem>
          <MenuItem>Action Two</MenuItem>
          <MenuItem>Action Three</MenuItem>
        </MenuList>
      </Menu>
    </Trigger>
  </div>
);
