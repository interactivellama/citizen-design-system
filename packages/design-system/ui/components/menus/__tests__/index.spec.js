// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.setup';
import { Submenu } from '../submenu/example';

const { matchesMarkupAndStyle } = createHelpers(__dirname);

it('renders a closed submenu', () => matchesMarkupAndStyle(<Submenu />));

it('renders an open submenu', () =>
  matchesMarkupAndStyle(<Submenu ariaExpanded />));
