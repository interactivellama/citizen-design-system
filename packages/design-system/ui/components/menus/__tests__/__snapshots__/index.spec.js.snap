// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Menu renders a base menu to the bottom 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
>
  <button
    aria-haspopup="true"
    className="slds-button slds-button_icon slds-button_icon-border-filled"
    id={undefined}
    title="Show More"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Show More
    </span>
  </button>
  <div
    className="slds-dropdown slds-dropdown_bottom"
    style={undefined}
  >
    <ul
      aria-label="Show More"
      className="slds-dropdown__list"
      role="menu"
    >
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="0"
        >
          <span
            className="slds-truncate"
            title="Menu Item One"
          >
            Menu Item One
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Two"
          >
            Menu Item Two
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Three"
          >
            Menu Item Three
          </span>
        </a>
      </li>
      <li
        className="slds-has-divider_top-space"
        role="separator"
      />
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Four"
          >
            Menu Item Four
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Base Menu renders a base menu to the left 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
>
  <button
    aria-haspopup="true"
    className="slds-button slds-button_icon slds-button_icon-border-filled"
    id={undefined}
    title="Show More"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Show More
    </span>
  </button>
  <div
    className="slds-dropdown slds-dropdown_left"
    style={undefined}
  >
    <ul
      aria-label="Show More"
      className="slds-dropdown__list"
      role="menu"
    >
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="0"
        >
          <span
            className="slds-truncate"
            title="Menu Item One"
          >
            Menu Item One
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Two"
          >
            Menu Item Two
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Three"
          >
            Menu Item Three
          </span>
        </a>
      </li>
      <li
        className="slds-has-divider_top-space"
        role="separator"
      />
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Four"
          >
            Menu Item Four
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Base Menu renders a base menu to the right 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
>
  <button
    aria-haspopup="true"
    className="slds-button slds-button_icon slds-button_icon-border-filled"
    id={undefined}
    title="Show More"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Show More
    </span>
  </button>
  <div
    className="slds-dropdown slds-dropdown_right"
    style={undefined}
  >
    <ul
      aria-label="Show More"
      className="slds-dropdown__list"
      role="menu"
    >
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="0"
        >
          <span
            className="slds-truncate"
            title="Menu Item One"
          >
            Menu Item One
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Two"
          >
            Menu Item Two
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Three"
          >
            Menu Item Three
          </span>
        </a>
      </li>
      <li
        className="slds-has-divider_top-space"
        role="separator"
      />
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Four"
          >
            Menu Item Four
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Menu with submenu renders a closed submenu 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
>
  <button
    aria-haspopup="true"
    className="slds-button slds-button_icon slds-button_icon-border-filled"
    id={undefined}
    title="Show More"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Show More
    </span>
  </button>
  <div
    className="slds-dropdown slds-dropdown_left"
    style={undefined}
  >
    <ul
      aria-label="Show More"
      className="slds-dropdown__list"
      role="menu"
    >
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="0"
        >
          <span
            className="slds-truncate"
            title="Menu Item One"
          >
            Menu Item One
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Two"
          >
            Menu Item Two
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item slds-has-submenu"
        role="presentation"
      >
        <a
          aria-expanded={undefined}
          aria-haspopup="true"
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Three"
          >
            Menu Item Three
          </span>
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small slds-icon-text-default slds-m-left_small slds-shrink-none"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#right"
            />
          </svg>
        </a>
        <div
          className="slds-dropdown slds-dropdown_submenu slds-dropdown_submenu-right"
          style={undefined}
        >
          <ul
            aria-label="Menu Item Three"
            className="slds-dropdown__list"
            role="menu"
          >
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item One"
                >
                  Submenu Item One
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Two"
                >
                  Submenu Item Two
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Three"
                >
                  Submenu Item Three
                </span>
              </a>
            </li>
            <li
              className="slds-has-divider_top-space"
              role="separator"
            />
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Four"
                >
                  Submenu Item Four
                </span>
              </a>
            </li>
          </ul>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Menu with submenu renders an open submenu 1`] = `
<div
  className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
>
  <button
    aria-haspopup="true"
    className="slds-button slds-button_icon slds-button_icon-border-filled"
    id={undefined}
    title="Show More"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Show More
    </span>
  </button>
  <div
    className="slds-dropdown slds-dropdown_left"
    style={undefined}
  >
    <ul
      aria-label="Show More"
      className="slds-dropdown__list"
      role="menu"
    >
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="0"
        >
          <span
            className="slds-truncate"
            title="Menu Item One"
          >
            Menu Item One
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item"
        role="presentation"
      >
        <a
          aria-checked={undefined}
          className={undefined}
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Two"
          >
            Menu Item Two
          </span>
        </a>
      </li>
      <li
        className="slds-dropdown__item slds-has-submenu"
        role="presentation"
      >
        <a
          aria-expanded={true}
          aria-haspopup="true"
          href="javascript:void(0);"
          role="menuitem"
          tabIndex="-1"
        >
          <span
            className="slds-truncate"
            title="Menu Item Three"
          >
            Menu Item Three
          </span>
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small slds-icon-text-default slds-m-left_small slds-shrink-none"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#right"
            />
          </svg>
        </a>
        <div
          className="slds-dropdown slds-dropdown_submenu slds-dropdown_submenu-right"
          style={undefined}
        >
          <ul
            aria-label="Menu Item Three"
            className="slds-dropdown__list"
            role="menu"
          >
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item One"
                >
                  Submenu Item One
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Two"
                >
                  Submenu Item Two
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Three"
                >
                  Submenu Item Three
                </span>
              </a>
            </li>
            <li
              className="slds-has-divider_top-space"
              role="separator"
            />
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                aria-checked={undefined}
                className={undefined}
                href="javascript:void(0);"
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Submenu Item Four"
                >
                  Submenu Item Four
                </span>
              </a>
            </li>
          </ul>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
